js-intro.js

```diff
@@ -47,16 +81,62 @@ var testString = "thisisateststring"
 
 // 5a. Write a function that takes the variable as an argument and returns the string without vowels. Expected output: "thsstststrng"
 
+function vowels(testString){
+    let vowels = "aeiou"
+    let arrV = []
+    for(let i = 0; i < testString.length; i++){
+        if(!vowels.includes(testString[i])){
+            arrV.push(testString[i])
+        }
+    }
+    return arrV.join('')
+}
+
+console.log(vowels(testString))
 // 5b. Stretch: Update your function to return an error message if the variable is not a string. Copy and paste your function to add these changes.
+// function error(testString){
+//     for(let i = 0; i < testString.length; i++){
+//         if(testString !== typeof("string")){
+//              return "error"
+//     }else{
+//         return "is a string"
+// }
+// }
+// console.log(error(testString))
 
 
 // ----------------------------------------------
 // Consider the following variable:
 
-var toonimals = [ {name: "Itchy", animal: "mouse"}, {name: "Stimpy", animal: "cat"}, {name: "Daffy", animal: "duck"}, {name: "Scratchy", animal: "cat"}, {name: "Ren", animal: "dog"}, {name: "Felix", animal: "cat"}]
+var toonimals = [ {name: "Itchy", animal: "mouse"}, 
+                {name: "Stimpy", animal: "cat"}, 
+                {name: "Daffy", animal: "duck"}, 
+                {name: "Scratchy", animal: "cat"}, 
+                {name: "Ren", animal: "dog"}, 
+                {name: "Felix", animal: "cat"}]
 
 // 6a. Write a function that takes in the toonimals array as an argument and returns an array with only the toon objects that are cats. Expected output: [ { name: "Stimpy", animal: "cat" }, { name: "Scratchy", animal: "cat" }, { name: "Felix", animal: "cat" } ]
-
+function cats(cat){
+    let arrC = []
+    for (let i = 0; i < cat.length; i++){
+    if(cat[i].animal === "cat"){
+        arrC.push(cat[i])
+    }
+}   return arrC
+}    
+console.log(cats(toonimals))
 //6b. Write a function that returns only the names of all the non-cats. Expected output: "Itchy" "Daffy" "Ren"
```

Comment:
This looks great! Here you passed in an argument as a placeholder, then called your function with the toonimals variable. Use this as a model to refactor your other functions.